firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -104,22 +104,31 @@ public interface InstanceRegistry {
   private final GrpcMetadataProvider metadataProvider;
 
   @NonNull
-  public static FirebaseFirestore getInstance() {
+  private static FirebaseApp getDefaultFirebaseApp() {
     FirebaseApp app = FirebaseApp.getInstance();
     if (app == null) {
       throw new IllegalStateException("You must call FirebaseApp.initializeApp first.");
     }
-    return getInstance(app, DatabaseId.DEFAULT_DATABASE_ID);
+    return app;
+  }
+
+  @NonNull
+  public static FirebaseFirestore getInstance() {
```

Comment:
Hmm, it looks like we never had public doc for `getInstance` for Java, can you add them now since it gets slightly more complex?

Reply:
Documentation added.